### PR TITLE
Add reply buttons interactive element and ability to send a message as a reply to previous message 

### PR DESCRIPTION
**Pull request by johnflash4real**

### CODE REVIEW

File: src/Client.php [hunk 1]
```diff
@@ -42,11 +42,11 @@ public function __construct(string $graph_version, ?ClientHandler $handler = nul
      *
      * @throws Netflie\WhatsAppCloudApi\Response\ResponseException
      */
-    public function sendMessage(Request\RequestWithBody $request): Response
+    public function sendMessage(Request\RequestWithBody $request,  array $extraData=[]): Response
     {
         $raw_response = $this->handler->postJsonData(
             $this->buildRequestUri($request->nodePath()),
-            $request->body(),
+            $request->body() + $extraData,
```
Comment (aalbarca) [inline src/Client.php:49]: We need to retrieve the body from the request. This approach adds complexity.

File: src/WhatsAppCloudApi.php [hunk 2]
```diff
@@ -51,6 +51,21 @@ public function __construct(array $config)
...
 
+    protected ?string $contextMessageId = null;
+
+    /**
+     * Send the next message as a reply to the supplied message id
+     *
+     * @param $messageId
+     * @return $this
+     */
+    public function asReplyTo(string $messageId): WhatsAppCloudApi
```
Comment (aalbarca) [inline src/WhatsAppCloudApi.php:62]: This superclass is not using Fluent API code style to simplify the code and isolate the responsibilities of the classes.

File: src/WhatsAppCloudApi.php [hunk 3]
```diff
@@ -359,4 +387,16 @@ public function fromPhoneNumberId(): string
     {
         return $this->app->fromPhoneNumberId();
     }
+
+    protected function getMsgRequestExtraData(): array
```
Comment (aalbarca) [inline src/WhatsAppCloudApi.php:391]: We need to get build the request with this extra data without create extra arguments on the Client class.

**Review by aalbarca — CHANGES_REQUESTED** [review]
Hey!

Thanks a lot for your awesome code. I added some comments.

If you need some type of help let me know.

Saludos.

File: src/WhatsAppCloudApi.php (hunk 3 above)
Comment (derrickobedgiu1) [reply src/WhatsAppCloudApi.php:391]: Context ID parameters should be optional. There seems to be a an issue with Graph from 26th June, 2023. Some reply buttons and lists won't get sent, and surprisingly they won't identify the issue. Also whenever I try to report the issue and share the explorer session the issue can't be submitted with a notice to wait and try again later.

The only workaround which is hurting though is to not pass in the context id if you're replying to any message id. It's not an overall issue but having the context id mandatory isn't something I would go with.

Overall, these seem to be the long-awaited missing features of this lib

**Comment by ianrothmann** [comment]
Hi everyone,

Thanks for your great work on this. Any chance we can resolve the final issues and get this merged?

Kind regards,

Ian

**Comment by aalbarca** [comment]
Feature to "reply to" whatsapp messages added in https://github.com/netflie/whatsapp-cloud-api/pull/146

**Pull request closed without merging**